lib/rules/selector-pseudo-element-colon-notation/index.mjs

```diff
@@ -48,43 +48,45 @@ const rule = (primary, _secondaryOptions, context) => {
 				return;
 			}
 
-			const fixedSelector = parseSelector(selector, result, ruleNode, (selectors) => {
-				selectors.walkPseudos((pseudo) => {
-					const pseudoElement = pseudo.value.replaceAll(':', '');
-
-					if (!levelOneAndTwoPseudoElements.has(pseudoElement.toLowerCase())) {
-						return;
-					}
-
-					const isDouble = pseudo.value.startsWith('::');
-
-					if (primary === 'single' && !isDouble) {
-						return;
-					}
-
-					if (primary === 'double' && isDouble) {
-						return;
-					}
-
-					if (context.fix) {
-						pseudo.replaceWith(pseudo.clone({ value: fixedColon + pseudoElement }));
-
-						return;
-					}
-
-					report({
-						message: messages.expected(primary),
-						node: ruleNode,
-						index: pseudo.sourceIndex,
-						endIndex: pseudo.sourceIndex + (isDouble ? 2 : 1),
-						result,
-						ruleName,
-					});
+			const selectorRoot = parseSelector(selector, result, ruleNode);
```

Comment:
question

Can't you use `?.walkPseudos(`?

Reply:
Code you suggested has the same effect, right? I think early return is more understandable.